client/mysqlbinlog.cc

```diff
@@ -3212,6 +3298,12 @@ int main(int argc, char** argv)
     fprintf(result_file, "/*!50530 SET @@SESSION.PSEUDO_SLAVE_MODE=0*/;\n");
   }
 
+  if (progress_file != NULL)
+  {
+    fprintf(progress_file, "EXIT\n");
```

Comment:
Is this the only `EXIT` path out of the program?

Reply:
You're right. There seems to be more exit() in the program that we may need to print `EXIT` log. Please see my reply under Robert's comment.